ui/desktop/app/controllers/scopes/scope/projects/sessions/index.js

```diff
@@ -4,25 +4,59 @@
  */
 
 import Controller from '@ember/controller';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
 import orderBy from 'lodash/orderBy';
+import { statusTypes } from 'api/models/session';
 
 export default class ScopesScopeProjectsSessionsIndexController extends Controller {
+  // =services
+
   // =attributes
 
+  queryParams = [{ targets: { type: 'array' } }];
+
+  @tracked targets = [];
+
+  // =methods
+
+  /**
+   * A list of sessions sorted by created time
+   * @type {[SessionModel]}
+   */
+  get sortedSessions() {
+    return orderBy(this.model.sessions, 'created_time', 'desc');
+  }
+
+  /**
+   * Returns targets that are associated will all sessions the user has access to
+   * with a limit of 100 targets rendered.
+   * @returns {[TargetModel]}
+   */
+  get availableTargets() {
+    const uniqueSessionTargetIds = new Set(
+      this.model.allSessions.map((session) => session.target_id),
+    );
+    return this.model.allTargets
+      .filter((target) => uniqueSessionTargetIds.has(target.id))
+      .slice(0, 100);
```

Comment:
I think you should do this in a separate PR, we should be doing this in the drop down component but this will also prevent you from searching all possible targets right?

Reply:
Ah yup you're correct, and good point I'll remove this